internal/jobobject/jobobject.go

```diff
@@ -91,140 +88,102 @@ func Create(ctx context.Context, name string, notifications bool) (_ *JobObject,
 		}
 	}()
 
+	job := &JobObject{
+		handle: jobHandle,
+	}
+
 	// If the IOCP we'll be using to receive messages for all jobs hasn't been
 	// created, create it and start polling.
 	if notifications {
-		ioInitOnce.Do(func() {
-			h, err := windows.CreateIoCompletionPort(windows.InvalidHandle, 0, 0, 0xffffffff)
-			if err != nil {
-				initIOErr = err
-				return
-			}
-			ioCompletionPort = h
-			go pollIOCP(ctx, h)
-		})
-
-		if initIOErr != nil {
-			return nil, initIOErr
-		}
-
-		mq = queue.NewMessageQueue()
-		jobMap.Store(uintptr(jobHandle), mq)
-		if err = attachIOCP(jobHandle, ioCompletionPort); err != nil {
-			jobMap.Delete(uintptr(jobHandle))
+		mq, err := setupNotifications(ctx, job)
+		if err != nil {
 			return nil, err
 		}
+		job.mq = mq
 	}
 
-	return &JobObject{
-		handle: jobHandle,
-		mq:     mq,
-	}, nil
+	return job, nil
 }
 
-// SetResourceLimits sets resource limits on the job object (cpu, memory, storage).
-func (job *JobObject) SetResourceLimits(limits *JobLimits) error {
-	// Go through and check what limits were specified and apply them to the job.
-	if limits.MemoryLimitInBytes != 0 {
-		if err := job.SetMemoryLimit(limits.MemoryLimitInBytes); err != nil {
-			return errors.Wrap(err, "failed to set job object memory limit")
-		}
+// Open opens an existing job object with name `name`.
+//
+// `notifications` specifies if the job will be registered to receive notifications.
+// If this is false, `PollNotifications` will return immediately with error `errNotRegistered`.
+//
+// Returns a JobObject structure and an error if there is one.
+func Open(ctx context.Context, name string, notifications bool) (_ *JobObject, err error) {
+	if name == "" {
+		return nil, errors.New("no job object name specified to open")
 	}
-
-	if limits.CPULimit != 0 {
-		if err := job.SetCPULimit(RateBased, limits.CPULimit); err != nil {
-			return errors.Wrap(err, "failed to set job object cpu limit")
-		}
-	} else if limits.CPUWeight != 0 {
-		if err := job.SetCPULimit(WeightBased, limits.CPUWeight); err != nil {
-			return errors.Wrap(err, "failed to set job object cpu limit")
-		}
+	unicodeJobName, err := winapi.NewUnicodeString(name)
+	if err != nil {
+		return nil, err
 	}
 
-	if limits.MaxBandwidth != 0 || limits.MaxIOPS != 0 {
-		if err := job.SetIOLimit(limits.MaxBandwidth, limits.MaxIOPS); err != nil {
-			return errors.Wrap(err, "failed to set io limit on job object")
-		}
+	var oa winapi.ObjectAttributes
+	oa.Length = unsafe.Sizeof(winapi.ObjectAttributes{})
+	oa.ObjectName = uintptr(unsafe.Pointer(unicodeJobName))
+	oa.Attributes = 0
+
+	var jobHandle windows.Handle
+	status := winapi.NtOpenJobObject(&jobHandle, winapi.JOB_OBJECT_ALL_ACCESS, &oa)
```

Comment:
Since `Create` uses `CreateJobObject`, but `Open` uses `NtOpenJobObject`, this will result in them not being compatible, since the `Nt` variant doesn't prepend the base named object directory to the path. Seems like we should instead take an option in both `Create` and `Open` to instruct if we should use the `Nt` function or not.

Reply:
Done, I added a new 'OptionsJobObjects` struct to match behavior for other container/pod creations.